### ISSUE TITLE
`doc_paragraphs_missing_punctuation` triggers on trailing emoji

### ISSUE THREAD

**Issue by BenjaminBrienen** [rust-lang/rust-clippy#16482]
### Summary

Emojis do not need to be punctuated. Rather, I would consider them to be punctuation.

### Lint Name

doc_paragraphs_missing_punctuation

### Reproducer

I tried this code:

```rust
/// A hasher designed to eke a little more speed out, given `TypeId`'s known characteristics.
///
/// Specifically, this is a no-op hasher that expects to be fed a u64's worth of
/// randomly-distributed bits. It works well for `TypeId` (eliminating start-up time, so that my
/// `get_missing` benchmark is ~30ns rather than ~900ns, and being a good deal faster after that, so
/// that my `insert_and_get_on_260_types` benchmark is ~12μs instead of ~21.5μs), but will
/// panic in debug mode and always emit zeros in release mode for any other sorts of inputs, so
/// yeah, do not use it! 😀
#[derive(Default)]
pub struct TypeIdHasher {
    value: u64,
}
```

I saw this happen:

```
warning: doc paragraphs should end with a terminal punctuation mark
  --> crates\stdx\src\anymap.rs:31:27
   |
31 | /// yeah, do not use it! 😀
   |                            ^ help: end the paragraph with some punctuation: `.`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.93.0/index.html#doc_paragraphs_missing_punctuation
```

I expected to see this happen:
No lint

### Version

```text
rustc 1.93.0 (254b59607 2026-01-19)
binary: rustc
commit-hash: 254b59607d4417e9dffbc307138ae5c86280fe4c
commit-date: 2026-01-19
host: x86_64-pc-windows-msvc
release: 1.93.0
LLVM version: 21.1.8
```
